modules/storage-api/src/testFixtures/java/org/apache/ignite/internal/storage/AbstractMvTableStorageTest.java

```diff
@@ -311,51 +313,89 @@ public void testMisconfiguredIndices() {
 
     @Test
     public void testStartRebalanceMvPartition() throws Exception {
+        assertThrows(StorageException.class, () -> tableStorage.startRebalanceMvPartition(PARTITION_ID_1));
+
+        UUID hashIndexId = hashIdx.id();
+        UUID sortedIndexId = sortedIdx.id();
+
         MvPartitionStorage partitionStorage = tableStorage.getOrCreateMvPartition(PARTITION_ID);
+        HashIndexStorage hashIndexStorage = tableStorage.getOrCreateHashIndex(PARTITION_ID, hashIndexId);
+        SortedIndexStorage sortedIndexStorage = tableStorage.getOrCreateSortedIndex(PARTITION_ID, sortedIndexId);
+
+        RowId rowId = new RowId(PARTITION_ID);
+
+        TestKey key = new TestKey(0, "0");
+
+        BinaryRow binaryRow = binaryRow(key, new TestValue(1, "1"));
+        IndexRowImpl indexRow = new IndexRowImpl(keyValueBinaryTuple(binaryRow), rowId);
 
         partitionStorage.runConsistently(() -> {
-            partitionStorage.addWriteCommitted(
-                    new RowId(PARTITION_ID),
-                    binaryRow(new TestKey(0, "0"), new TestValue(1, "1")),
-                    clock.now()
-            );
+            partitionStorage.addWriteCommitted(rowId, binaryRow, clock.now());
 
             partitionStorage.lastAppliedIndex(100);
 
             return null;
         });
 
+        hashIndexStorage.put(indexRow);
+
+        sortedIndexStorage.put(indexRow);
+
         partitionStorage.flush().get(1, TimeUnit.SECONDS);
 
         tableStorage.startRebalanceMvPartition(PARTITION_ID).get(1, TimeUnit.SECONDS);
 
         MvPartitionStorage newPartitionStorage0 = tableStorage.getMvPartition(PARTITION_ID);
+        HashIndexStorage newHashIndexStorage0 = tableStorage.getOrCreateHashIndex(PARTITION_ID, hashIndexId);
+        SortedIndexStorage newSortedIndexStorage0 = tableStorage.getOrCreateSortedIndex(PARTITION_ID, sortedIndexId);
 
         assertNotNull(newPartitionStorage0);
+        assertNotNull(newHashIndexStorage0);
+        assertNotNull(newSortedIndexStorage0);
+
         assertNotSame(partitionStorage, newPartitionStorage0);
+        assertNotSame(hashIndexStorage, newHashIndexStorage0);
```

Comment:
Why do you assert things like these? They look completely wrong. What if we have a partition listener that has a cached value of MvPartitionStorage? Should it invalidate it and ask for another one?
Old instances must work. Please fix the test and the test implementation

Reply:
Here we check that all storages are new and empty.
Why is this wrong?

Reply:
Fix it